### PR TITLE
feat: Add DataPlane https injestion for SecOps exporter

### PR DESCRIPTION
**Pull request by shazlehu**
### Proposed Change

Adds support for the new DataPlane REST injestion API for SecOps.

Refer to bindplane PR https://github.com/observIQ/bindplane-op-enterprise/pull/3443 for testing setup.

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: exporter/chronicleexporter/exporter.go [hunk 1]
```diff
@@ -90,23 +96,39 @@ func newExporter(cfg *Config, params exporter.Settings, collectorID, exporterID
...
+			return nil, fmt.Errorf("dial: %w", err)
+		}
+
+		exp.conn = conn
+		exp.client = api.NewIngestionServiceV2Client(conn)
+
+		if cfg.CollectAgentMetrics {
+			exp.wg.Add(1)
+			go exp.startHostMetricsCollection(ctx)
+		}
```
[the remark this answers is not in the document]
Comment (shazlehu) [reply exporter/chronicleexporter/exporter.go:119]: Specific to gRPC for now

File: exporter/chronicleexporter/marshal.go [hunk 2]
```diff
@@ -224,3 +225,81 @@ func (m *protoMarshaler) constructPayloads(rawLogs map[string][]*api.LogEntry) [
...
+
+func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) []*api.ImportLogsRequest {
+	payloads := make([]*api.ImportLogsRequest, 0, len(rawLogs))
+
+	// TODO logType
+	for _, entries := range rawLogs {
+		if len(entries) > 0 {
+			payloads = append(payloads,
+				&api.ImportLogsRequest{
+					// TODO: Add parent and hint
```
[the remark this answers is not in the document]
Comment (shazlehu) [reply exporter/chronicleexporter/marshal.go:290]: We still haven't received clarity on what these fields are for, so this TODO seems appropriate to me

File: exporter/chronicleexporter/marshal_test.go [hunk 3]
```diff
@@ -192,6 +192,170 @@ func TestProtoMarshaler_MarshalRawLogs(t *testing.T) {
...
+		// 	labels: []*api.Label{},
+		// 	logRecords: func() plog.Logs {
+		// 		return mockLogs(mockLogRecord("Log with overridden type", map[string]any{"chronicle_log_type": "ASOC_ALERT"}))
+		// 	},
+		// 	expectations: func(t *testing.T, requests []*api.ImportLogsRequest) {
+		// 		require.Len(t, requests, 1)
+		// 		batch := requests[0].Batch
+		// 		require.Equal(t, "ASOC_ALERT", batch.LogType, "Expected log type to be overridden by attribute")
+		// 	},
+		// },
```
[the remark this answers is not in the document]
Comment (shazlehu) [reply exporter/chronicleexporter/marshal_test.go:338]: Yup, these tests don't apply here, deleted them

File: exporter/chronicleexporter/marshal.go [hunk 4]
```diff
@@ -224,3 +225,81 @@ func (m *protoMarshaler) constructPayloads(rawLogs map[string][]*api.LogEntry) [
...
+
+func buildForwarderString(cfg Config) string {
+	format := "projects/%s/locations/%s/instances/%s/forwarders/%s"
+	return fmt.Sprintf(format, cfg.Project, cfg.Location, cfg.CustomerID, cfg.Forwarder)
+}
+
+func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) []*api.ImportLogsRequest {
+	payloads := make([]*api.ImportLogsRequest, 0, len(rawLogs))
+
+	// TODO logType
```
Comment (dpaasman00) [inline exporter/chronicleexporter/marshal.go:285]: does this logType need to be handled yet or is this another thing that needs clarity?

**Review by dpaasman00 — CHANGES_REQUESTED** [review]
Looks good and is working well for me. 

Minor nit to add a more informative comment to lingering TODOs about why we're waiting/what we're waiting for

File: exporter/chronicleexporter/marshal.go (hunk 4 above)
Comment (shazlehu) [reply exporter/chronicleexporter/marshal.go:285]: It should be handled now, *however*, there's a bug in the way this is being handled in that's addressed in my other open agent PR. Currently, it won't respect the source parameter for log type, so you need the attribute `log_type` set to `windows_event.security` (that's what Windows Events will set it to)

**Review by dpaasman00 — APPROVED** [review]

**Pull request merged**